### PR TITLE
Wire `hab-pkg-export-docker` into `hab` & setup for building.

### PR DESCRIPTION
**Pull request by fnichol**
There are a few changes in this PR, split out to make things a bit clearer.

* The removal of old functional testing setup was done when searching the source tree for references to `dockerize`.
* A hab-only change to update the CLI delegating code to invoke the new Rust program directly and avoid the existing common `hab pkg export` logic.
* Add the package into the build and release process, driven primarily by `.travis.yml`.

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by fnichol** [comment]
I'm going to take a quick swing at compiling on macOS to confirm the Mac behavior (warn and quit).

File: components/hab/src/command/pkg/export/docker.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    const EXPORT_CMD_ENVVAR: &'static str = "HAB_PKG_EXPORT_DOCKER_BINARY";
+    const EXPORT_PKG_IDENT: &'static str = "core/hab-pkg-export-docker";
+    const EXPORT_PKG_IDENT_ENVVAR: &'static str = "HAB_PKG_EXPORT_DOCKER_PKG_IDENT";
+
+    pub fn start(ui: &mut UI, args: Vec<OsString>) -> Result<()> {
+        let command = match henv::var(EXPORT_CMD_ENVVAR) {
+            Ok(command) => PathBuf::from(command),
+            Err(_) => {
+                init();
```
Comment (reset) [inline components/hab/src/command/pkg/export/docker.rs]: Should we be starting this earlier in the function? I think I made changes recently so this wouldn't panic if you called it more than once - can't remember exactly

Comment (fnichol) [reply components/hab/src/command/pkg/export/docker.rs]: We could do a future refactor which just jams that early in `hab`. We need this for the install logic/verify code and have been only calling it when necessary. But yeah, I've never been 100% happy with that. It's possible that the overhead of that call is so small that it isn't an issue.

**Comment by fnichol** [comment]
I should note that for the moment, I left `components/pkg-dockerize` in the source tree as I'm not 100% sure if it's being depended upon by any other software. We should be good to remove that in a few weeks unless we learn anything.

**Comment by fnichol** [comment]
Took me a couple of tries on my rebuilt macOS to make sure I had all linked library deps. Looks good though.

tenor-97992166

@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/279789719).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/279789719).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**